starter_code/queries.md

```diff
@@ -4,80 +4,72 @@
 
 ### 1. All the companies whose name match 'Babelgum'. Retrieve only their `name` field.
 
-<!-- Your Code Goes Here -->
+filter : { name : "Babelgum" }
+project : { name : 1 }
 
 ### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
-
-<!-- Your Code Goes Here -->
+filter : { number_of_employees: { $gt: 5000 } }
+sort : { number_of_employees: 1 }
+limit : 20
 
 ### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fields.
-
-<!-- Your Code Goes Here -->
+filter : { $and: [ { founded_year: { $gte: 2000 } }, { founded_year: { $lte: 2005} } ] }
+project : {name: 1, founded_year: 1}
 
 ### 4. All the companies that had a Valuation Amount of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
-
-<!-- Your Code Goes Here -->
+filter : { $and: [ {"ipo.valuation_amount": { $gte: "100000000" } }, { founded_year: { $lt: "2010" } } ] }
+project : { name: 1, ipo: 1}
 
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
-
-<!-- Your Code Goes Here -->
+filter : { $and: [ { number_of_employees: { $lt: 1000 } }, { founded_year: {$lt: 2005} } ] }
+sort : { number_of_employees: -1}
 
 ### 6. All the companies that don't include the `partners` field.
-
-<!-- Your Code Goes Here -->
+filter : {partners: {$exists: false}}
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+filter : { category_code: { $in: [ null ] } }
```

Comment:
It's working but you could make it simpler  this way 
{ category_code: null}
or 
{ category_code: {$eq: null}}